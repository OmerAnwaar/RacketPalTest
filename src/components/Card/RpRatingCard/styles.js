import {StyleSheet} from 'react-native';
import {colors, theme} from '../../../shared/constants';

const styles = StyleSheet.create({
  starContainer: {
    height: 242,
    borderRadius: 12,
    backgroundColor: colors.WHITE,
  },
  containerWidth: {width: '90%'},

  subContainer: {
    ...theme.SINGLE_FLEX,
    alignItems: 'center',
    justifyContent: 'space-evenly',
    paddingHorizontal: '5%',
  },
  editIcon: {
    height: 47,
    width: 47,
  },
  starTitle: {
    fontSize: 16,
    paddingTop: '15%',
    fontWeight: '700',
    textAlign: 'center',
  },
  editContainer: {
    top: -20,
    width: 84,
    height: 84,
    ...theme.CENTER,
    borderRadius: 45,
    position: 'absolute',
    backgroundColor: colors.WHITE,
  },
  subTitleStar: {
    fontSize: 12,
    lineHeight: 16.8,
    fontWeight: '400',
    textAlign: 'center',
    paddingHorizontal: 40,
  },
  label: {
    fontSize: 10,
    lineHeight: 15,
    fontWeight: '700',
    textAlign: 'left',
    color: colors.PRIMARY,
  },
  remindmelater: {
    fontSize: 10,
    lineHeight: 15,
    fontWeight: '700',
    textAlign: 'center',
    paddingHorizontal: 40,
    textTransform: 'uppercase',
    color: colors.NOBEL,
  },
  starSubContainer: {
    paddingVertical: 10,
    width: 300,
    justifyContent: 'space-evenly',
  },
  inputStyle: {
    width: '100%',
    borderWidth: 1,
    textAlignVertical: 'top',
    borderColor: colors.PRIMARY,
    borderRadius: 6,
    height: 112,
    padding: '3%',
    paddingTop: '3%',
  },
  submitButton: {width: '100%', backgroundColor: colors.PRIMARY},
});

export default styles;
