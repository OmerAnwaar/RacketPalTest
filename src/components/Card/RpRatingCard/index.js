import React from 'react';
import {
  Image,
  KeyboardAvoidingView,
  Platform,
  Pressable,
  Text,
  TextInput,
  View,
} from 'react-native';
import {AirbnbRating} from 'react-native-ratings';
import {colors, icons} from '../../../shared/constants';
import styles from './styles';
import RpButton from '../../RpButton';

const RpRatingCard = ({
  rating,
  onDismiss,
  handleCancel,
  onRatingCompleted,
  onSubmit,
  onRemindMeLater,
}) => {
  return (
    <KeyboardAvoidingView
      behavior={Platform.OS === 'ios' ? 'padding' : 'height'}>
      <Pressable onPress={onDismiss} style={styles.containerWidth}>
        <View style={[styles.starContainer, rating && {height: 400}]}>
          <View style={styles.subContainer}>
            <View style={styles.editContainer}>
              <Image
                source={icons.EDIT}
                style={styles.editIcon}
                resizeMode={'contain'}
              />
            </View>
            <Text style={styles.starTitle}>Enjoying RacketPal?</Text>
            <Text style={styles.subTitleStar}>
              Tap a star to rate it on the App store
            </Text>
            <AirbnbRating
              size={30}
              count={5}
              showRating={false}
              defaultRating={0}
              selectedColor={colors.PRIMARY}
              onFinishRating={onRatingCompleted}
              starContainerStyle={styles.starSubContainer}
            />
            {rating < 4 && rating !== 0 ? (
              <>
                <Text style={styles.label}>ANY FEEDBACK FOR US?</Text>
                <TextInput
                  style={styles.inputStyle}
                  selectionColor={colors.PRIMARY}
                  multiline={true}
                />
                <RpButton
                  label={'Submit'}
                  style={styles.submitButton}
                  onPress={onSubmit}
                />
              </>
            ) : (
              <Pressable onPress={onRemindMeLater}>
                <Text style={styles.remindmelater}>remind me later</Text>
              </Pressable>
            )}
          </View>
        </View>
      </Pressable>
    </KeyboardAvoidingView>
  );
};
export default RpRatingCard;
