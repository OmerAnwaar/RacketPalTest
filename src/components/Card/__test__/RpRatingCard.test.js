import React from 'react';
import renderer from 'react-test-renderer';
import RpRatingCard from '../RpRatingCard';

describe('RpRatingCard Component', () => {
  it('renders RpRatingCard Component correctly', () => {
    const snapshot = renderer.create(<RpRatingCard />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
