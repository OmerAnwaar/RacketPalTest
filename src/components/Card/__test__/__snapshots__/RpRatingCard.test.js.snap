// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RpRatingCard Component renders RpRatingCard Component correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "paddingBottom": 0,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "width": "90%",
      }
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "white",
            "borderRadius": 12,
            "height": 242,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "space-evenly",
            "paddingHorizontal": "5%",
          }
        }
      >
        <View
          style={
            {
              "alignContent": "center",
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 45,
              "height": 84,
              "justifyContent": "center",
              "position": "absolute",
              "top": -20,
              "width": 84,
            }
          }
        >
          <Image
            resizeMode="contain"
            source={
              {
                "testUri": "../../../src/assets/Icons/ic_edit.png",
              }
            }
            style={
              {
                "height": 47,
                "width": 47,
              }
            }
          />
        </View>
        <Text
          style={
            {
              "fontSize": 16,
              "fontWeight": "700",
              "paddingTop": "15%",
              "textAlign": "center",
            }
          }
        >
          Enjoying RacketPal?
        </Text>
        <Text
          style={
            {
              "fontSize": 12,
              "fontWeight": "400",
              "lineHeight": 16.8,
              "paddingHorizontal": 40,
              "textAlign": "center",
            }
          }
        >
          Tap a star to rate it on the App store
        </Text>
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flexDirection": "column",
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                {
                  "justifyContent": "space-evenly",
                  "paddingVertical": 10,
                  "width": 300,
                },
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Image
                collapsable={false}
                source={
                  {
                    "testUri": "../../react-native-ratings/dist/images/airbnb-star.png",
                  }
                }
                style={
                  {
                    "height": 30,
                    "margin": 3,
                    "tintColor": "#BDC3C7",
                    "transform": [
                      {
                        "scale": 1,
                      },
                    ],
                    "width": 30,
                  }
                }
              />
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Image
                collapsable={false}
                source={
                  {
                    "testUri": "../../react-native-ratings/dist/images/airbnb-star.png",
                  }
                }
                style={
                  {
                    "height": 30,
                    "margin": 3,
                    "tintColor": "#BDC3C7",
                    "transform": [
                      {
                        "scale": 1,
                      },
                    ],
                    "width": 30,
                  }
                }
              />
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Image
                collapsable={false}
                source={
                  {
                    "testUri": "../../react-native-ratings/dist/images/airbnb-star.png",
                  }
                }
                style={
                  {
                    "height": 30,
                    "margin": 3,
                    "tintColor": "#BDC3C7",
                    "transform": [
                      {
                        "scale": 1,
                      },
                    ],
                    "width": 30,
                  }
                }
              />
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Image
                collapsable={false}
                source={
                  {
                    "testUri": "../../react-native-ratings/dist/images/airbnb-star.png",
                  }
                }
                style={
                  {
                    "height": 30,
                    "margin": 3,
                    "tintColor": "#BDC3C7",
                    "transform": [
                      {
                        "scale": 1,
                      },
                    ],
                    "width": 30,
                  }
                }
              />
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Image
                collapsable={false}
                source={
                  {
                    "testUri": "../../react-native-ratings/dist/images/airbnb-star.png",
                  }
                }
                style={
                  {
                    "height": 30,
                    "margin": 3,
                    "tintColor": "#BDC3C7",
                    "transform": [
                      {
                        "scale": 1,
                      },
                    ],
                    "width": 30,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Text
            style={
              {
                "color": "#B4B4B4",
                "fontSize": 10,
                "fontWeight": "700",
                "lineHeight": 15,
                "paddingHorizontal": 40,
                "textAlign": "center",
                "textTransform": "uppercase",
              }
            }
          >
            remind me later
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
